metaflow/plugins/kfp/tests/run_integration_tests.py

```diff
@@ -51,10 +58,53 @@ def obtain_flow_file_paths(flow_dir_path: str) -> List[str]:
         and not file_name.startswith(".")
         and not "raise_error_flow" in file_name
         and not "accelerator_flow" in file_name
+        and not "s3_sensor_flow" in file_name
+        and not "upload_to_s3_flow" in file_name
     ]
     return file_paths
 
 
+def test_s3_sensor_flow(pytestconfig) -> None:
+    # ensure the s3_sensor waits for some time before the key exists
+    random_string = ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(15))
+    file_name = f"{random_string}.txt"
+
+    upload_to_s3_flow_cmd = f"{_python()} flows/upload_to_s3_flow.py --datastore=s3 kfp run "
+    s3_sensor_flow_cmd = f"{_python()} flows/s3_sensor_flow.py --datastore=s3 kfp run --wait-for-completion "
+
+    main_config_cmds = (
+        f"--workflow-timeout 1800 "
+        f"--max-parallelism 3 --experiment metaflow_test --tag test_t1 "
```

Comment:
why is there a max-parallelism of 3, does this run in parallel?

Reply:
This is something to explore in another PR - yes the tests run in parallel. I'm thinking of tweaking this parameter in the future to help resolve issues like https://zbrt.atl.zillow.net/browse/AIP-4523.

Reply:
This KFP CLI parameter controls `Maximum number of parallel pods.` I believe you actually made this change a while back to limit resources usage by the flows so they didn't go beyond the limits sets in the Metaflow integ testing namespace.

Reply:
But is this flow running anything in parallel? If not we can remove, else it's adding added coded that isn't doing anything to this test confusing the reader